src/core/agents.jl

```diff
@@ -221,3 +211,231 @@ macro agent(struct_repr)
         end
     quote Base.@__doc__($(esc(expr))) end
 end
+
+"""
+    @compact struct YourCompactedAgentType{X,Y}(BaseAgentType) [<: OptionalSupertype]
```

Comment:
```suggestion
    @compact struct YourParentAgentType{X,Y}(BaseAgentType) [<: OptionalSupertype]
```

perhaps parent is a better way to refer to this.